iothub_client/src/iothub_message.c

```diff
@@ -213,57 +265,36 @@ IOTHUB_MESSAGE_HANDLE IoTHubMessage_Clone(IOTHUB_MESSAGE_HANDLE iotHubMessageHan
         }
         else
         {
-            result->messageId = NULL;
-            result->correlationId = NULL;
-            result->userDefinedContentType = NULL;
-            result->contentEncoding = NULL;
+            InitializeIoTHubMessage(result);
```

Comment:
If you want to initialize the new fields to 0, just add the two to the list. Better make it explicit in the function

Reply:
done